packages/@aws-cdk/aws-apigateway/test/base-path-mapping.test.ts

```diff
@@ -77,6 +101,78 @@ describe('BasePathMapping', () => {
     }).toThrowError(/base path may only contain/);
   });
 
+  test('throw error for basePath starting with /', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const api = new apigw.RestApi(stack, 'MyApi');
+    api.root.addMethod('GET'); // api must have atleast one method.
+    const domain = new apigw.DomainName(stack, 'MyDomain', {
+      domainName: 'example.com',
+      certificate: acm.Certificate.fromCertificateArn(stack, 'cert', 'arn:aws:acm:us-east-1:1111111:certificate/11-3336f1-44483d-adc7-9cd375c5169d'),
+      endpointType: apigw.EndpointType.REGIONAL,
+    });
+
+    // WHEN
+    const invalidBasePath = '/invalid-base-path';
+
+    // THEN
+    expect(() => {
+      new apigw.BasePathMapping(stack, 'MyBasePath', {
+        restApi: api,
+        domainName: domain,
+        basePath: invalidBasePath,
+      });
+    }).toThrowError(/A base path cannot start or end with/);
+  });
+
+  test('throw error for basePath ending with /', () => {
```

Comment:
APIGateway definitely forbids this?

Reply:
Verified it via the AWS Console. Screenshot attached below - 
![Screenshot 2022-12-16 at 1 18 33 AM](https://user-images.githubusercontent.com/7585096/207953609-d4e6febc-1751-4f1a-a342-395ead04952d.png)